hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AbstractMessageTask.java

```diff
@@ -185,8 +185,12 @@ private void checkPermissions(ClientEndpoint endpoint) {
     protected abstract void processMessage() throws Throwable;
 
     protected void sendResponse(Object response) {
-        ClientMessage clientMessage = encodeResponse(response);
-        sendClientMessage(clientMessage);
+        try {
+            ClientMessage clientMessage = encodeResponse(response);
+            sendClientMessage(clientMessage);
+        } catch (Exception e) {
+            sendClientMessage(e);
```

Comment:
don't you want to log the exception on `finest` here?

Reply:
Yes, make sense. I have added a call to `logProcessingFailure`.